Src/Newtonsoft.Json.UnityConverters/Random/RandomStateConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using State = UnityEngine.Random.State;
+
+namespace Newtonsoft.Json.UnityConverters.Random
+{
+    public class RandomStateConverter : PartialIntConverter<State>
+    {
+        private static readonly string[] _memberNames = { "s0", "s1", "s2", "s3" };
+
+        private static readonly FieldInfo[] _fieldInfos = _memberNames
```

Comment:
Nullable right?
This needs some sort of validation that it has the fields we expect before we use them.